src/data/W/instances.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2015 Joseph Hua. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Hua
+-/
+import data.W.basic
+
+/-!
+# Instances of W types
+
+We take the view of W types as inductive types.
+Given `α : Type` and `β : α → Type`, the W type determined by this data, `W_type β`, is the
+inductively with constructors from `α` and arities of each constructor `a : α` given by `β a`.
+
+This file contains `nat` and `list` as examples of W types.
```

Comment:
Can you add something like
```suggestion
This file contains `nat` and `list` as examples of W types.

## Main results

* `W_type.nat_equiv_W_type`: some summary mentioning `W_type.nat_β`
* `W_type.list_equiv_W_type`: some summary mentioning `W_type.list_β`
```

This also raises that `W_type.list_equiv_W_type` probably doesn't need to mention `W_type` twice.